tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,149 @@
+// <copyright file="ServerlessMiniAgent.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation;
+
+internal static class ServerlessMiniAgent
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ServerlessMiniAgent));
+
+    internal static string GetMiniAgentPath(PlatformID os, ImmutableTracerSettings settings)
+    {
+        if (!settings.IsRunningInGCPFunctions && !settings.IsRunningInAzureFunctionsConsumptionPlan)
+        {
+            return null;
+        }
+
+        if (Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH") != null)
+        {
+            return Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH");
+        }
+
+        // Environment.OSVersion.Platform can return PlatformID.Unix on MacOS, this is OK as GCP & Azure don't have MacOs functions.
+        if (Environment.OSVersion.Platform != PlatformID.Unix && Environment.OSVersion.Platform != PlatformID.Win32NT)
+        {
+            Log.Error("Serverless Mini Agent is only supported on Windows and Linux.");
+            return null;
+        }
+
+        string rustBinaryPathRoot;
+        if (settings.IsRunningInGCPFunctions)
+        {
+            rustBinaryPathRoot = "/layers/google.dotnet.publish/publish/bin";
+        }
+        else
+        {
+            rustBinaryPathRoot = "/home/site/wwwroot";
+        }
+
+        var isWindows = os == PlatformID.Win32NT;
+
+        string rustBinaryPathOsFolder = isWindows ? "datadog-serverless-agent-windows-amd64" : "datadog-serverless-agent-linux-amd64";
+        string rustBinaryName = string.Format("datadog-serverless-trace-mini-agent{0}", isWindows ? ".exe" : string.Empty);
```

Comment:
Consider this instead to avoid calling `string.Format()`:
```suggestion
        string rustBinaryName = isWindows ? "datadog-serverless-trace-mini-agent.exe" : "datadog-serverless-trace-mini-agent");
```